server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesReader.java

```diff
@@ -32,11 +57,159 @@
  */
 @ExperimentalApi
 public class Composite99DocValuesReader extends DocValuesProducer implements CompositeIndexReader {
-    private DocValuesProducer delegate;
+    private static final Logger logger = LogManager.getLogger(Composite99DocValuesReader.class);
+
+    private final DocValuesProducer delegate;
+    private IndexInput dataIn;
+    private ChecksumIndexInput metaIn;
+    private final Map<String, IndexInput> compositeIndexInputMap = new LinkedHashMap<>();
+    private final Map<String, CompositeIndexMetadata> compositeIndexMetadataMap = new LinkedHashMap<>();
+    private final List<String> fields;
+    private DocValuesProducer compositeDocValuesProducer;
+    private final List<CompositeIndexFieldInfo> compositeFieldInfos = new ArrayList<>();
+    private SegmentReadState readState;
 
-    public Composite99DocValuesReader(DocValuesProducer producer, SegmentReadState state) throws IOException {
+    public Composite99DocValuesReader(DocValuesProducer producer, SegmentReadState readState) throws IOException {
         this.delegate = producer;
-        // TODO : read star tree files
+        this.fields = new ArrayList<>();
+
+        String metaFileName = IndexFileNames.segmentFileName(
+            readState.segmentInfo.name,
+            readState.segmentSuffix,
+            Composite99DocValuesFormat.META_EXTENSION
+        );
+
+        String dataFileName = IndexFileNames.segmentFileName(
+            readState.segmentInfo.name,
+            readState.segmentSuffix,
+            Composite99DocValuesFormat.DATA_EXTENSION
+        );
+
+        boolean success = false;
+        try {
+
+            // initialize meta input
+            dataIn = readState.directory.openInput(dataFileName, readState.context);
+            CodecUtil.checkIndexHeader(
+                dataIn,
+                Composite99DocValuesFormat.DATA_CODEC_NAME,
+                Composite99DocValuesFormat.VERSION_START,
+                Composite99DocValuesFormat.VERSION_CURRENT,
+                readState.segmentInfo.getId(),
+                readState.segmentSuffix
+            );
+
+            // initialize data input
+            metaIn = readState.directory.openChecksumInput(metaFileName, readState.context);
+            Throwable priorE = null;
+            try {
+                CodecUtil.checkIndexHeader(
+                    metaIn,
+                    Composite99DocValuesFormat.META_CODEC_NAME,
+                    Composite99DocValuesFormat.VERSION_START,
+                    Composite99DocValuesFormat.VERSION_CURRENT,
+                    readState.segmentInfo.getId(),
+                    readState.segmentSuffix
+                );
+
+                while (true) {
+
+                    // validate magic marker
+                    long magicMarker = metaIn.readLong();
+                    if (magicMarker == -1) {
+                        logger.info("EOF reached for composite index metadata");
```

Comment:
Lets remove this

Reply:
moved this to debug

Reply:
I don't think we need to put it to debug as well, once we enable debug logs, this will pollute the logs which it did in my benchmarks quite a bit  .